source/Octopus.Client/OctopusAsyncRepository.cs

```diff
@@ -134,5 +142,118 @@ public OctopusAsyncRepository(IOctopusAsyncClient client)
         public IWorkerRepository Workers { get; }
         public IScopedUserRoleRepository ScopedUserRoles { get; }
         public IUserPermissionsRepository UserPermissions { get; }
+
+        private SpaceRootResource SpaceRootDocument { get; set; }
+        private RootResource RootDocument { get; set; }
+
+        public async Task<bool> HasLink(string name)
+        {
+            await LoadRootDocuments();
```

Comment:
Anywhere where you've introduced an `await` within this project, you should also use `.ConfigureAwait(false)`